mpt/core/src/main/java/org/apache/james/mpt/session/ExternalSessionFactory.java

```diff
@@ -34,7 +35,7 @@
 public class ExternalSessionFactory implements SessionFactory {
 
     public static final String IMAP_SHABANG = "* OK IMAP4rev1 Server ready";
-    protected final InetSocketAddress address;
+    protected InetSocketAddress address;
```

Comment:
Mutability here smells bad.

I would like to keep this immutable, even if it means extracting James start fro, the 'host system'

Reply:
I updated the fixup, please check https://github.com/linagora/james-project/pull/1640/commits/88a15a6f6d875144930fef3489490d40a98807a0